### ISSUE TITLE
Double NOT operator in IF statement's expression

### ISSUE THREAD

**Issue by kangax** [jshint/jshint#1487]
`if (!!smth) { ... }` should be warned against, as redundant

**Comment by stanislavgeorgiev** [comment]
`undefined === false // result: false
!!undefined === false // result: true`

The double bang can be very useful for converting `null` or `undefined` to `false`

**Comment by eliperelman** [comment]
@stanislavgeorgiev that may be the case, but this issue is about whether that has value while inside an `if` condition (as answered above, no, it doesn't). These statements are equivalent:

```js
if (undefined)
```

```js
if (!!undefined === false)
```

```js
if (!!undefined)
```

If you were passing outside of a condition, maybe as input to JSON.stringify, then it would have potential value:

```js
JSON.stringify(!!undefined) // 'false'
JSON.stringify({ value: !!undefined === false }) // '{"value":true}'
```

**Comment by ljharb** [comment]
@eliperelman also answered above, it does have style value - it clearly conveys the programmer's awareness of the boolean coercions.

**Comment by eliperelman** [comment]
@ljharb good point, I guess I was only thinking about it from the evaluation side. :+1:
